### DISCUSSION TITLE
with-redux-thunk state is lost with dynamic routes

### DISCUSSION THREAD

**Discussion by tanmoyAtb** [vercel/next.js#16545] [Help] [answered]
I tried to reuse the **with-redux-thunk** example.
in these pages
`pages/[pageId]/index.js`
`pages/[pageId]/show-redux-state.js`

Looks like redux state is lost in dynamic routes.

**Comment by rokinsky** [comment, marked as answer]
@tanmoytb this should work fine, check if your pages are reloaded during transitions - you should use `next/link` correctly (with `as` and `href` props) to prevent reloading.
